### ISSUE TITLE
Support multiple reorganization rulesets

### ISSUE THREAD

**Issue by MelGrubb** [codecadwallader/codemaid#532]
### Environment
- Visual Studio version: 2017 Pro
- CodeMaid version: 10.4
- Code language: C#

### Description
Not all classes want to be organized in the same way. For instance, test fixture classes may have a specific order that makes more sense such as Given(), When(), Then(). Another example might be a code-first migration where Up() typically comes before Down().

I can see two ways to differentiate rules. The first would be for each set of rules to have a regex that "recognizes" classes. The second would be an attribute name that the rule applies to. The first isn't definitive enough, but the second should do just fine since both of the cases I can think of where the ordering rules would be different should be expected to have an attribute on the class. Either way, I think a first-match-wins is probably fine, as long as the rulesets can be rearranged relative to each other. Then, I could put the "catch-all" ruleset at the bottom of the list to mop any leftovers.

### Steps to recreate
1. Reorganize any class, note that the same rules apply to every class.

### Current behavior
All classes get reorganized the same way.

### Expected behavior
Specialized classes should have their own rule sets.

**Comment by codecadwallader** [comment]
Thanks for the suggestion.  There's some overlap with #317 but it sounds like you want to take it further and allow for custom rules (not just disabling in certain scenarios).

**Comment by MelGrubb** [comment]
Exactly. Not to draw comparisons, but I was able to accomplish this with ReSharper through the use of attributes to differentiate which rules applies to which classes. That's great and all, but it's not very useful for open-source projects, or on-site consulting where I can't count on all the developers having access to the same tools. CodeMaid, being an free and open-source project, means that I can specify it as a requirement on a particular development team... apart from certain state agencies having very restrictive tooling policies and review boards (ugh, don't get me started).
